aws/resource_aws_api_gateway_vpc_link_test.go

```diff
@@ -0,0 +1,132 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAwsAPIGatewayVpcLink_basic(t *testing.T) {
+	rName := acctest.RandString(5)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsAPIGatewayVpcLinkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAPIGatewayVpcLinkConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAPIGatewayVpcLinkExists("aws_api_gateway_vpc_link.test"),
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "name", fmt.Sprintf("tf-apigateway-%s", rName)),
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "description", "test"),
+				),
+			},
+			{
+				Config: testAccAPIGatewayVpcLinkConfig_Update(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAPIGatewayVpcLinkExists("aws_api_gateway_vpc_link.test"),
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "name", fmt.Sprintf("tf-apigateway-update-%s", rName)),
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "description", "test update"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsAPIGatewayVpcLinkDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_api_gateway_vpc_link" {
+			continue
+		}
+
+		input := &apigateway.GetVpcLinkInput{
+			VpcLinkId: aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.GetVpcLink(input)
+		if err != nil {
+			if isAWSErr(err, apigateway.ErrCodeNotFoundException, "") {
+				return nil
+			}
+			return err
+		}
+
+		if *resp.Status != apigateway.VpcLinkStatusDeleting {
```

Comment:
I think we should treat any positive response here as a failure - the VPC link should be already deleted by the time test finishes and practically have no _status_ (i.e. it shouldn't be in `deleting` status).